File: v4-client-js/examples/transfer_example_deposit.ts
import Long from 'long';

import { BECH32_PREFIX, IDeposit } from '../src';
import { Network } from '../src/clients/constants';
import LocalWallet from '../src/clients/modules/local-wallet';
import { TransactionMsg, TransactionType } from '../src/clients/modules/post';
import { SubaccountInfo } from '../src/clients/subaccount';
import { ValidatorClient } from '../src/clients/validator-client';
import { DYDX_TEST_MNEMONIC } from './constants';

async function test(): Promise<void> {
  const wallet = await LocalWallet.fromMnemonic(DYDX_TEST_MNEMONIC, BECH32_PREFIX);
  console.log(wallet);

  const client = await ValidatorClient.connect(Network.testnet().validatorConfig);
  console.log('**Client**');
  console.log(client);

  const subaccount = new SubaccountInfo(wallet, 0);
  const tx = await client.post.deposit(
    subaccount,
    0,
    new Long(10_000_000),
  );
  console.log('**Deposit Tx**');
  console.log(tx);

  const params: IDeposit = {
    quantums: new Long(10_000_000),
    assetId: 0,
  };
  const tx2 = await client.post.sendMsg(
    subaccount,
    new TransactionMsg(
      TransactionType.DEPOSIT,
      params,
      false,
    ),
  );
  console.log('**Deposit Tx**');
  console.log(tx2);
}

test().then(() => {
}).catch((error) => {
  console.log(error.message);
});

File: v4-client-py-v2/examples/transfer_example_deposit.py
import asyncio

from dydx_v4_client.network import TESTNET
from dydx_v4_client.node.client import NodeClient
from dydx_v4_client.node.message import subaccount
from dydx_v4_client.node.subaccount import SubaccountInfo
from dydx_v4_client.wallet import Wallet
from tests.conftest import DYDX_TEST_MNEMONIC, TEST_ADDRESS


async def test():
    # Create the client
    node = await NodeClient.connect(TESTNET.node)

    # Create the wallet
    wallet = await Wallet.from_mnemonic(node, DYDX_TEST_MNEMONIC, TEST_ADDRESS)
    subaccount_info = SubaccountInfo.for_wallet(wallet, 0)

    # Call the API - deposit funds
    response = await node.deposit(
        subaccount_info, TEST_ADDRESS, subaccount(TEST_ADDRESS, 0), 0, 10_000_000
    )
    print(response)


asyncio.run(test())
